From 11001f7a2060e9662983df9dbe67a4a7f8045fa7 Mon Sep 17 00:00:00 2001
From: Philip Zeyliger <philip@cloudera.com>
Date: Fri, 20 Nov 2015 19:43:41 +0000
Subject: [PATCH 1018/2848] CLOUDERA-BUILD Specify assembly artifact naming.

Reason: New Feature
Author: Philip Zeyliger
Ref: CDH-34616

When using SNAPSHOT artifacts, there are two names
for an hbase jar.  For example,
  hbase-server-1.0.0-cdh5.7.6-20151201.205646-1.jar (stored in
  org/apache/hbase/hbase-server/1.0.0-cdh5.7.6-SNAPSHOT in a maven repo)
or
  hbase-server-1.0.0-cdh5.7.6-SNAPSHOT.jar

Depending on whether you're using the local ~/.m2/repository or a "remote"
file:// repository (and equivalently using "mvn install" vs "mvn deploy"), the
assembly gets different names.  Our world expects the -SNAPSHOT name, and fails
subtly if it doesn't happen.

The suble error you'd see would be something like this from
the RPM build:

    [DEBUG] adding entry hadoop-2.6.0-cdh5.7.6-SNAPSHOT/share/hadoop/mapreduce/sources/hadoop-mapreduce-client-app-2.6.0-cdh5.7.6-SNAPSHOT-sources.jar
    ...
    [DEBUG] adding entry hadoop-2.6.0-cdh5.7.6-SNAPSHOT/share/hadoop/mapreduce/hadoop-mapreduce-client-app-2.6.0-cdh5.7.6-20151120.212535-3.jar
    ...
    + '[' -e /mnt/philip/cdh-trunk/build/cdh5/hadoop/2.6.0-cdh5.7.6-SNAPSHOT/rpm/BUILDROOT/hadoop-2.6.0+cdh5.7.6+981-1.cdh5.7.6.p0.el6.x86_64/usr/lib/hadoop-mapreduce//hadoop-mapreduce-client-app-2.6.0-cdh5.7.6-SNAPSHOT.jar ']'
    + exit 1

(Those DEBUG lines were created by adding -X to the mvn invocation in do-component-build,
which was how I got to the bottom of it.)

This very small XML change makes that error go away.

Other people have experienced this at https://issues.apache.org/jira/browse/MASSEMBLY-67.

Change-Id: I3dd47f2c558fd9914ea6f3389e5f0684a16a338f
---
 .../resources/assemblies/hadoop-mapreduce-dist.xml |    3 +++
 .../main/resources/assemblies/hadoop-yarn-dist.xml |    4 ++++
 2 files changed, 7 insertions(+)

diff --git a/hadoop-assemblies/src/main/resources/assemblies/hadoop-mapreduce-dist.xml b/hadoop-assemblies/src/main/resources/assemblies/hadoop-mapreduce-dist.xml
index 749e16a..3c3df50 100644
--- a/hadoop-assemblies/src/main/resources/assemblies/hadoop-mapreduce-dist.xml
+++ b/hadoop-assemblies/src/main/resources/assemblies/hadoop-mapreduce-dist.xml
@@ -163,6 +163,7 @@
         <outputDirectory>share/hadoop/${hadoop.component}</outputDirectory>
         <includeDependencies>false</includeDependencies>
         <unpack>false</unpack>
+        <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       </binaries>
     </moduleSet>
     <moduleSet>
@@ -175,6 +176,7 @@
         <outputDirectory>share/hadoop/${hadoop.component}</outputDirectory>
         <includeDependencies>false</includeDependencies>
         <unpack>false</unpack>
+        <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       </binaries>
     </moduleSet>
   </moduleSets>
@@ -182,6 +184,7 @@
     <dependencySet>
       <useProjectArtifact>false</useProjectArtifact>
       <outputDirectory>/share/hadoop/${hadoop.component}/lib</outputDirectory>
+      <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       <!-- Exclude hadoop artifacts. They will be found via HADOOP* env -->
       <excludes>
         <exclude>org.apache.hadoop:hadoop-common</exclude>
diff --git a/hadoop-assemblies/src/main/resources/assemblies/hadoop-yarn-dist.xml b/hadoop-assemblies/src/main/resources/assemblies/hadoop-yarn-dist.xml
index a15e124..806c1bd 100644
--- a/hadoop-assemblies/src/main/resources/assemblies/hadoop-yarn-dist.xml
+++ b/hadoop-assemblies/src/main/resources/assemblies/hadoop-yarn-dist.xml
@@ -194,6 +194,7 @@
         <outputDirectory>share/hadoop/${hadoop.component}</outputDirectory>
         <includeDependencies>false</includeDependencies>
         <unpack>false</unpack>
+        <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       </binaries>
     </moduleSet>
     <moduleSet>
@@ -205,6 +206,7 @@
         <outputDirectory>share/hadoop/${hadoop.component}/test</outputDirectory>
         <includeDependencies>false</includeDependencies>
         <unpack>false</unpack>
+        <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       </binaries>
     </moduleSet>
   </moduleSets>
@@ -212,6 +214,7 @@
     <dependencySet>
       <useProjectArtifact>false</useProjectArtifact>
       <outputDirectory>/share/hadoop/${hadoop.component}/lib</outputDirectory>
+      <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       <!-- Exclude hadoop artifacts. They will be found via HADOOP* env -->
       <excludes>
         <exclude>org.apache.hadoop:*</exclude>
@@ -224,6 +227,7 @@
     <dependencySet>
       <useProjectArtifact>false</useProjectArtifact>
       <outputDirectory>/share/hadoop/${hadoop.component}/lib-examples</outputDirectory>
+      <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
       <includes>
         <include>org.hsqldb:hsqldb</include>
       </includes>
-- 
1.7.9.5

